sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpression.cs

```diff
@@ -6,14 +6,17 @@
 using System.Linq;
 using System.Text.Json.Serialization;
 using System.Collections.Generic;
+using System.Reflection;
 
 namespace Azure.Core.Expressions.DataFactory
 {
     /// <summary>
     /// A class representing either a primitive value or an expression.
     /// For details on DataFactoryExpressions see https://learn.microsoft.com/en-us/azure/data-factory/control-flow-expression-language-functions#expressions.
     /// </summary>
-    /// <typeparam name="T"> Can be one of <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="double"/>, <see cref="Array"/>, or <see cref="IList{TElement}"/>.</typeparam>
+    /// <typeparam name="T"> Can be one of <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="double"/>, <see cref="TimeSpan"/>,
+    /// <see cref="DateTimeOffset"/>, <see cref="Uri"/>, <see cref="IList{String}"/>, <see cref="IList{TElement}"/>,
+    /// or <see cref="IDictionary{String,String}"/>.</typeparam>
 #pragma warning disable SA1649 // File name should match first type name
     [JsonConverter(typeof(DataFactoryExpressionJsonConverter))]
     public sealed class DataFactoryExpression<T>
```

Comment:
Have we thought about mockability here?

I am ok if that is a follow up feature but if we choose to do that can we create an issue to track?

Reply:
Ah I didn't even notice that this was sealed. I think it should be simple enough to actually create instances of this type for testing though rather than relying on mocks.